test_relint.py

```diff
@@ -7,18 +9,32 @@
 
 
 class TestMain:
-    def test_main_execution(self):
-        sys.argv.extend(['relint.py', 'test_relint.py'])
+    @pytest.mark.parametrize('filename', ['test_relint.py', '[a-b].py', '[b-a].py'])
+    def test_main_execution(self, filename, mocker):
+        mocker.patch.object(sys, 'argv', ['relint.py', filename])
+
         with pytest.raises(SystemExit) as exc_info:
             main()
 
         assert exc_info.value.code == 0
 
-    def test_main_execution_with_diff(self, capsys):
-        sys.argv.extend(['relint.py', 'test_relint.py', '--diff'])
-        sys.stdin = open('test.diff')
-        with pytest.raises(SystemExit) as exc_info:
-            main()
+    def test_main_execution_with_diff(self, capsys, mocker, tmpdir):
```

Comment:
I am not so sure about this change. I would definitely like to see this in a separate commit, since it's not related to the bug you described. Besides that, I do see your point. I don't know if we to mock so much, of if we couldn't just add to separate test fixture files here. A file that contains a line or two and a diff file.

Reply:
Moved test fixes into separate commit. And also simplified test a bit.